include/slb/utility.hpp

```diff
@@ -0,0 +1,356 @@
+  template <class T2, class T1>
+  constexpr T2&& get(pair<T1, T2> && p) noexcept;
+  template <class T2, class T1>
+  constexpr const T2&& get(const pair<T1, T2>&& p) noexcept;
+
+  // 23.4.5, pair piecewise construction
+  struct piecewise_construct_t {
+    explicit piecewise_construct_t() = default;
+  };
+  inline constexpr piecewise_construct_t piecewise_construct{};
+  template <class... Types>
+  class tuple; // defined in <tuple> (23.5.2)
+
+  // in-place construction
+  struct in_place_t {
+    explicit in_place_t() = default;
+  };
+  inline constexpr in_place_t in_place{};
+  template <class T>
+  struct in_place_type_t {
+    explicit in_place_type_t() = default;
+  };
+  template <class T>
+  inline constexpr in_place_type_t<T> in_place_type{};
+  template <size_t I>
+  struct in_place_index_t {
+    explicit in_place_index_t() = default;
+  };
+  template <size_t I>
+  inline constexpr in_place_index_t<I> in_place_index{};
+}
+
+*/
+
+#include <cstddef>
+#include <initializer_list>
+
+#include "detail/config.hpp"
+#include "detail/lib.hpp"
+#include "type_traits.hpp"
+
+namespace slb {
+
+// // 23.2.3, swap
+// template <class T>
+// void swap(T& a, T& b) noexcept(see below);
+// template <class T, size_t N>
+// void swap(T (&a)[N], T (&b)[N]) noexcept(is_nothrow_swappable_v<T>);
+
+// // 23.2.4, exchange
+// template <class T, class U = T>
+// T exchange(T& obj, U&& new_val);
+
+// // 23.2.5, forward/move
+// template <class T>
+// constexpr T&& forward(remove_reference_t<T>& t) noexcept;
+// template <class T>
+// constexpr T&& forward(remove_reference_t<T>&& t) noexcept;
+// template <class T>
+// constexpr remove_reference_t<T>&& move(T&&) noexcept;
+// template <class T>
+// constexpr conditional_t<!is_nothrow_move_constructible_v<T> &&
+//                             is_copy_constructible_v<T>,
+//                         const T&,
+//                         T&&>
+// move_if_noexcept(T& x) noexcept;
+
+// // 23.2.6, as_const
+// template <class T>
+// constexpr add_const_t<T>& as_const(T& t) noexcept;
+// template <class T>
+// void as_const(const T&&) = delete;
+
+// // 23.2.7, declval
+// template <class T>
+// add_rvalue_reference_t<T> declval() noexcept; // as unevaluated operand
+
+// 23.3, Compile-time integer sequences
+
+#if __cpp_lib_integer_sequence
+using std::integer_sequence;
+#else
+template <typename T, T... Is>
+struct integer_sequence {
+  static_assert(slb::is_integral<T>::value, "`T` shall be an integer type.");
+  using value_type = T;
+  static constexpr std::size_t size() noexcept { return sizeof...(Is); }
+};
+#endif
+
+template <std::size_t... Is>
+using index_sequence = slb::integer_sequence<std::size_t, Is...>;
+
+#if __has_builtin(__make_integer_seq) || defined(_MSC_VER)
+template <typename T, T N>
+using make_integer_sequence = __make_integer_seq<slb::integer_sequence, T, N>;
+#else
+namespace detail {
+
+template <typename T>
+struct make_integer_sequence_impl {
```

Comment:
Consider having a single implementation for `std::size_t`, and implement the other integer types by casting from it. That way, more of the generation machinery gets reused by different instantiations.